test/FunctionalTests/Microsoft.OData.Core.Tests/ScenarioTests/UriBuilder/FilterAndOrderByBuilderTests.cs

```diff
@@ -465,6 +465,14 @@ public void BuildFilterWithDollarItInFunctionCall()
             Uri actualUri = UriBuilder(queryUri, ODataUrlKeyDelimiter.Parentheses, settings);
             Assert.Equal(new Uri("http://gobbledygook/People(1)/RelatedSSNs?$filter=endswith%28%24it%2C%27xyz%27%29"), actualUri);
         }
+
+        [Fact]
+        public void BuildFilterWithManyBinaryOperator()
+        {
+            Uri queryUri = new Uri("People?$filter=ID eq 1 or ID eq 2 or ID eq 3 or ID eq 4 or ID eq 5", UriKind.Relative);

```

Comment:
>People?$filter=ID eq 1 or ID eq 2 or ID eq 3 or ID eq 4 or ID eq 5 [](start = 36, length = 66)

would be good to have a mix of and/or w/wout parens to validate the parens logic.